Documentation/app-operations.md

```diff
@@ -0,0 +1,225 @@
+3. Copy the service files of the application to the stage1 units directory `/usr/lib/systemd/system`.
+4. Start the injected application using the `app start` entrypoint.
+
+## rkt app eject
+
+`coreos.com/rkt/stage1/app/eject`
+
+1. receive a reference to an application that resides inside the pod (running or stopped)
+2. stop the application if it's running.
+3. remove the contents of the application (rootfs) from the pod (keep the logs?) and delete references to it (e.g. service files).
+   - Delegate these steps by creating `app/install` and `app/eject` entrypoints?
+
+The responsibility of this entrypoint is to receive an app inside a pod and to remove it from the pod.
+so that it is ready to be started by the stage1 supervisor (e.g. systemd for the default flavor).
+After this operation, starting the application again is not possible and requires injecting the same
+application using the `app/inject` entrypoint.
+
+### Stage1 Default Flavor Example Workflow
+
+1. Stop the application using the `app stop` entrypoint.
+2. Delete the application rootfs in `/var/lib/rkt/pods/run/$uuid/stage1/rootfs/opt/stage2/`.
+3. Remove the application service files.
+4. Perform garbage collection during the execution of a pod.
+5. Reload the systemd daemon for changes to take place without disturbing the other tennant applications.
+
+## rkt app start
+
+`coreos.com/rkt/stage1/app/start`
+
+1. given a reference to an application that's in a startable state (see below), instruct the supervisor to start the application.
+
+The responsibility of this entrypoint is to start an application that is in a `Ready` or `Stopped` state, 
+which might be an app that was recently injected or an app that was stopped, instruct the supervisor to start the application.
+
+## rkt app stop
+
+`coreos.com/rkt/stage1/app/stop`
+
+1. given a reference to an application that's in a stoppable state (see below), instruct the supervisor to stop the application.
+
+The responsibility of this entrypoint is to stop an application that is `Running` by instructing the stage1 supervisor. 
+
+**Note:** there is a graceful shutdown which sends the termination signal to application and waits for a grace period
+for application to terminate and then the application is shutdown if it doesn't terminate by the end of the grace period.
+
+**Discussion Point:** Should one-app-pods not be allowed to terminate when the app inside is stopped?
+At the moment, once the app stops the reaper terminates and the pod shuts down.
+
+# App Container States
+
+Expected set of application container states are listed below:
+
+```go
+type AppState string
+
+const (
+	// Ready to be started
+	ReadyAppState   AppState = "Ready"
+	StoppedAppState AppState = "Stop"
+
+	// Started
+	RunningAppState AppState = "Running"
+
+	FinishedAppState AppState = "Finished"
+
+	// No further operations can be done
+	EjectedAppState AppState = "Ejected"
```

Comment:
`Ejected` is redundant here, will remove it 👍 

Its a bit different than systemd unit states, because here we only care about the lifecycle state without any interest in success/failure. We can add intermediate states to here such as `Ejecting` for `Ejected`, `Finishing` for `Finished`.